### PR TITLE
[crypto] Gates batch verification under a feature

### PR DESCRIPTION
**Pull request by huitseeker**
This allows us to (de)-activate features relying on ed25519's "batch" if needed.

Workaround for #3567.

cc @n4ss 
review @kchalkias

### CODE REVIEW

**Comment by kchalkias** [comment]
@huitseeker let’s fix the linting issue

File: crypto/crypto/src/ed25519.rs [hunk 1]
```diff
@@ -359,6 +359,7 @@ impl Signature for Ed25519Signature {
         self.0.to_bytes().to_vec()
     }
 
+    #[cfg(batch)]
```
Comment (bmwill) [inline crypto/crypto/src/ed25519.rs:362]: Did you intend this to be the following instead?
```suggestion
    #[cfg(feature = "batch")]
```

**Comment by andll** [comment]
Looks like lint issue is not really lint issue - something wrong with compilation with this feature set

**Comment by andll** [comment]
Also, there is one more crate, `libra-network-address` imported from `libra-types` that imports `libra-crypto` without default_features=false

File: crypto/crypto/src/ed25519.rs (hunk 1 above)
Comment (huitseeker) [reply crypto/crypto/src/ed25519.rs:362]: :facepalm:

File: types/Cargo.toml [hunk 2]
```diff
@@ -28,7 +28,7 @@ thiserror = "1.0"
 tiny-keccak = { version = "2.0.2", default-features = false, features = ["sha3"] }
 
 lcs = { path = "../common/lcs", version = "0.1.0", package = "libra-canonical-serialization" }
-libra-crypto = { path = "../crypto/crypto", version = "0.1.0" }
+libra-crypto = { path = "../crypto/crypto", version = "0.1.0", default-features = false }
 libra-crypto-derive = { path = "../crypto/crypto-derive", version = "0.1.0" }
 libra-network-address = { path = "../network/network-address", version = "0.1.0" }
```
Comment (andll) [inline types/Cargo.toml:33]: Need default_features=false here as well

**Comment by andll** [comment]
Btw, aside from `libra-crypto` and `libra-types`, we also need `transaction-builder` and `move-core-types` crates.
They also import quite a bit of stuff recursively, do you think it could be simplier, if we just remove "batch" from defaults in libra-crypto, and only enable it on specific crates in libra that needs it?

**Comment by huitseeker** [comment]
@andll Yeah, the change in defaults clearly seems to be the right approach, due to feature unification only growing the feature set.

**Comment by thefallentree** [comment]
it would be great if we could wait until. the powerpc land-blocking test is added before merging this,

**Comment by andll** [comment]
@thefallentree  Why wait to land a fix?
I will check that this fix works manually, and I would actually prefer it merged soon, so I can switch to master commit

**Comment by huitseeker** [comment]
Matching branch on internal project is up.

**Pull request closed without merging**